services/stickers/transactions.go

```diff
@@ -84,63 +85,99 @@ func (api *API) Buy(ctx context.Context, chainID uint64, txArgs transactions.Sen
 	return tx.Hash().String(), nil
 }
 
-func (api *API) BuyEstimate(ctx context.Context, chainID uint64, from types.Address, packID *bigint.BigInt) (uint64, error) {
+func (api *API) BuyPrepareTxCallMsg(chainID uint64, from types.Address, packID *bigint.BigInt) (ethereum.CallMsg, error) {
 	callOpts := &bind.CallOpts{Context: api.ctx, Pending: false}
 
 	stickerType, err := api.contractMaker.NewStickerType(chainID)
 	if err != nil {
-		return 0, err
+		return ethereum.CallMsg{}, err
 	}
 
 	packInfo, err := stickerType.GetPackData(callOpts, packID.Int)
 	if err != nil {
-		return 0, err
+		return ethereum.CallMsg{}, err
 	}
 
 	stickerMarketABI, err := abi.JSON(strings.NewReader(stickers.StickerMarketABI))
 	if err != nil {
-		return 0, err
+		return ethereum.CallMsg{}, err
 	}
 
 	extraData, err := stickerMarketABI.Pack("buyToken", packID.Int, from, packInfo.Price)
 	if err != nil {
-		return 0, err
+		return ethereum.CallMsg{}, err
 	}
 
 	sntABI, err := abi.JSON(strings.NewReader(snt.SNTABI))
 	if err != nil {
-		return 0, err
+		return ethereum.CallMsg{}, err
 	}
 
 	stickerMarketAddress, err := stickers.StickerMarketContractAddress(chainID)
 	if err != nil {
-		return 0, err
+		return ethereum.CallMsg{}, err
 	}
 
 	data, err := sntABI.Pack("approveAndCall", stickerMarketAddress, packInfo.Price, extraData)
 	if err != nil {
-		return 0, err
-	}
-
-	ethClient, err := api.contractMaker.RPCClient.EthClient(chainID)
-	if err != nil {
-		return 0, err
+		return ethereum.CallMsg{}, err
 	}
 
 	sntAddress, err := snt.ContractAddress(chainID)
 	if err != nil {
-		return 0, err
+		return ethereum.CallMsg{}, err
 	}
 
-	return ethClient.EstimateGas(ctx, ethereum.CallMsg{
+	return ethereum.CallMsg{
 		From:  common.Address(from),
 		To:    &sntAddress,
 		Value: big.NewInt(0),
 		Data:  data,
-	})
+	}, nil
+}
+
+func (api *API) BuyPrepareTx(ctx context.Context, chainID uint64, from types.Address, packID *bigint.BigInt) (interface{}, error) {
```

Comment:
please review my changes, i've added this method, because we sign transaction in status-react